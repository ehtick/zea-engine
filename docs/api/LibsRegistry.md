<a name="LibsRegistry"></a>

### LibsRegistry
Libraries registry.



* [LibsRegistry](#LibsRegistry)
    * [new LibsRegistry(version)](#new-LibsRegistry)
    * [registerLib(packageJson)](#registerLib)
    * [listLibs() â‡’ <code>object</code>](#listLibs)

<a name="new_LibsRegistry_new"></a>

### new LibsRegistry
Construct a new libraries registry for the specific version.


| Param | Type | Description |
| --- | --- | --- |
| version | <code>string</code> | The version of the Zea Engine that will be validated against the registered libraries. |

<a name="LibsRegistry+registerLib"></a>

### registerLib
Validate and register a library.



| Param | Type | Description |
| --- | --- | --- |
| packageJson | <code>object</code> | The package.json of the library to register. |

<a name="LibsRegistry+listLibs"></a>

### listLibs
List the registered libraries with their versions.


**Returns**: <code>object</code> - Libraries list.  


### [Class Tests](api/./LibsRegistry.test)