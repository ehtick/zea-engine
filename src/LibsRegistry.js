/**
 * Libraries registry.
 */
class LibsRegistry {
  /**
   * Construct a new libraries registry for the specific version.
   * @param {string} version - The version of the Zea Engine that will be validated against the registered libraries.
   */
  constructor(version) {
    this.version = version
    this.registry = {}
  }

  /**
   * Validate and register a library.
   * @param {object} packageJson - The package.json of the library to register.
   */
  registerLib(packageJson) {
    const libName = packageJson.name
    const libVersion = packageJson.version

    this.registry[libName] = libVersion
    console.log(`Registered lib '${libName}' v${libVersion}`)
  }

  /**
   * List the registered libraries with their versions.
   * @return {object} Libraries list.
   */
  listLibs() {
    return this.registry
  }
}

export { LibsRegistry }
