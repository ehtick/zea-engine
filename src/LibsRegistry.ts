/**
 * Libraries registry.
 */
class LibsRegistry {
  registry: Record<string, unknown>
  version: string

  /**
   * Construct a new libraries registry for the specific version.
   * @param version - The version of the Zea Engine that will be validated against the registered libraries.
   */
  constructor(version: string) {
    this.version = version
    this.registry = {}
  }

  /**
   * Validate and register a library.
   * @param packageJson - The package.json of the library to register.
   */
  registerLib(packageJson: Record<string, any>): void {
    const libName = packageJson.name
    const libVersion = packageJson.version

    if (this.registry[libName]) {
      console.warn("Library already registered:" + libName)
    }

    this.registry[libName] = libVersion
    console.log(`Registered lib '${libName}' v${libVersion}`)
  }

  /**
   * List the registered libraries with their versions.
   * @return Libraries list.
   */
  listLibs(): Record<string, unknown> {
    return this.registry
  }
}

export { LibsRegistry }
