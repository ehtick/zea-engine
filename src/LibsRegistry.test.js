import { LibsRegistry } from './LibsRegistry'

const libsRegistry = new LibsRegistry('2.0.0')

describe('Libraries registry', () => {
  it('Registers a valid library', () => {
    const validPackageJson = {
      dependencies: {
        '@zeainc/zea-engine': '^2.0.0',
      },
      name: 'fake-lib',
      version: '0.0.1',
    }

    libsRegistry.registerLib(validPackageJson)

    expect(libsRegistry.listLibs()).toMatchSnapshot()
  })
})
