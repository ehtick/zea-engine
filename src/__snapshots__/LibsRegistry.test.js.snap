// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Libraries registry Registers a valid library 1`] = `
Object {
  "fake-lib": "0.0.1",
}
`;
